librosa/feature/spectral.py

```diff
@@ -939,6 +942,90 @@ def chroma_cqt(y=None, sr=22050, C=None, hop_length=512, fmin=None,
     return chroma
 
 
+def chroma_cens(y=None, sr=22050, C=None, hop_length=512, fmin=None,
+               threshold=0.001, tuning=None, n_chroma=12,
+               n_octaves=7, bins_per_octave=None, cqt_mode='full', window=None,
+               win_len_smooth=41, downsample_smooth=10):
+    r'''CENS-chromagram
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,)]
+        audio time series
+
+    sr : number > 0
+        sampling rate of `y`
+
+    C : np.ndarray [shape=(d, t)] [Optional]
+        a pre-computed constant-Q spectrogram
+
+    hop_length : int > 0
+        number of samples between successive chroma frames
+
+    Returns
+    -------
+    chroma_cens : np.ndarray [shape=(n_chroma, t)]
+        The output cens-chromagram
+
+    See Also
+    --------
+
+    Examples
+    --------
+
+    '''
+
+    cqt_func = {'full': cqt, 'hybrid': hybrid_cqt}
+
+    if bins_per_octave is None:
+        bins_per_octave = n_chroma
+
+    # Build the CQT if we don't have one already
+    if C is None:
+        C = np.abs(cqt_func[cqt_mode](y, sr=sr,
+                                      hop_length=hop_length,
+                                      fmin=fmin,
+                                      n_bins=n_octaves * bins_per_octave,
+                                      bins_per_octave=bins_per_octave,
+                                      tuning=tuning,
+                                      real=False))
+
+    chroma = chroma_cqt(C=C, norm=None,
+                        fmin=fmin,
+                        bins_per_octave=bins_per_octave,
+                        n_octaves=n_octaves)
+
+    # L1-Normalization
+    chroma = util.normalize(chroma, norm=1, axis=0,
+                            mode='fill_constant', threshold=threshold)
+
+    # Quantize amplitudes
+    quant_steps = [0.4, 0.2, 0.1, 0.05]
+    quant_weights = [0.25, 0.25, 0.25, 0.25]
+
+    chroma_quant = np.zeros_like(chroma)
+
+    for cur_quant_step_idx, cur_quant_step in enumerate(quant_steps):
+        chroma_quant += (chroma > cur_quant_step) * quant_weights[cur_quant_step_idx]
+
+    # Apply temporal smoothing
+    win = scipy.signal.hann(win_len_smooth)
```

Comment:
In scipy at least, `hanning` is an alias for `hann`, but not to be confused with `hamming`.  Is matlab different?



> I won't change it in the implementation, but doing the unit-testing would mean that I would test the reference Chroma-Toolbox-CENS with 41 tap smoothing window against a 43 tap librosa implementation.

This seems reasonable.